### PR TITLE
KFSTP-1690 - fix CINV Total Billed To Date amount...

### PR DESCRIPTION
**Pull request by bhutchinson**
to reflect total Award information and be recalculated as necessary

### CODE REVIEW

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java [hunk 1]
```diff
@@ -1133,18 +1119,45 @@ protected void populateContractsGrantsInvoiceDocument(ContractsAndGrantsBillingA
...
+    }
 
-
-        // To set the New Total Billed Amount.
-        if (document.getInvoiceMilestones().size() > 0) {
-            document.getInvoiceGeneralDetail().setNewTotalBilled(totalMilestoneAmount);
-        }
-        else if (document.getInvoiceBills().size() > 0) {
-            document.getInvoiceGeneralDetail().setNewTotalBilled(totalBillAmount);
+    private KualiDecimal calculateTotalExpenditureAmount(ContractsGrantsInvoiceDocument document) {
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: protected, por favor?

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java [hunk 2]
```diff
@@ -1122,7 +1105,10 @@ protected void populateContractsGrantsInvoiceDocument(ContractsAndGrantsBillingA
             }
         }
         totalMilestoneAmount = totalMilestoneAmount.add(document.getInvoiceGeneralDetail().getBilledToDateAmount());
+        return totalMilestoneAmount;
+    }
 
+    private KualiDecimal calculateTotalBillAmount(ContractsGrantsInvoiceDocument document) {
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: protected

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java [hunk 3]
```diff
@@ -1080,38 +1080,21 @@ protected void populateContractsGrantsInvoiceDocument(ContractsAndGrantsBillingA
...
-        totalExpendituredAmount = totalExpendituredAmount.add(document.getInvoiceGeneralDetail().getBilledToDateAmount());
-
+        else {
+            newTotalBilled = calculateTotalExpenditureAmount(document);
+            newTotalBilled = getContractsGrantsInvoiceDocumentService().getOtherNewTotalBilledForAwardPeriod(document);
+        }
+        document.getInvoiceGeneralDetail().setNewTotalBilled(newTotalBilled);
+    }
 
+    private KualiDecimal calculateMilestoneAmount(ContractsGrantsInvoiceDocument document) {
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: protected

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java [hunk 4]
```diff
@@ -385,8 +386,36 @@ public void recalculateNewTotalBilled(ContractsGrantsInvoiceDocument contractsGr
...
+    }
+
+    /**
+     * @see org.kuali.kfs.module.ar.document.service.ContractsGrantsInvoiceDocumentService#getOtherNewTotalBilledForAwardPeriod(org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument)
+     */
+    @Override
+    public KualiDecimal getOtherNewTotalBilledForAwardPeriod(ContractsGrantsInvoiceDocument contractsGrantsInvoiceDocument) {
+        KualiDecimal newTotalBilled = KualiDecimal.ZERO;
+
+        Map<String, String> fieldValuesForInvoice = new HashMap<String, String>();
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: just fyi, HashMap<>() would have worked...just to save extra typing next time

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java [hunk 5]
```diff
@@ -113,8 +113,8 @@ public void prepareForSave() {
         super.prepareForSave();
         // To do a recalculate of current expenditures in invoice details section so that the totals get affected properly.
 
-        // To be performed whenever the document is saved only for awards without Milestones, Bills or LOC Billing
-        if (!this.getInvoiceGeneralDetail().getBillingFrequencyCode().equalsIgnoreCase(ArConstants.MILESTONE_BILLING_SCHEDULE_CODE) && !this.getInvoiceGeneralDetail().getBillingFrequencyCode().equalsIgnoreCase(ArConstants.PREDETERMINED_BILLING_SCHEDULE_CODE) && !this.getInvoiceGeneralDetail().getBillingFrequencyCode().equalsIgnoreCase(ArConstants.LOC_BILLING_SCHEDULE_CODE)) {
+        // To be performed whenever the document is saved only for awards without Milestones or Bills
+        if (!this.getInvoiceGeneralDetail().getBillingFrequencyCode().equalsIgnoreCase(ArConstants.MILESTONE_BILLING_SCHEDULE_CODE) && !this.getInvoiceGeneralDetail().getBillingFrequencyCode().equalsIgnoreCase(ArConstants.PREDETERMINED_BILLING_SCHEDULE_CODE)) {
```
Comment (smith750) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: Should we be using StringUtils.equalsIgnoreCase here?

File: work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java (hunk 4 above)
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/service/impl/ContractsGrantsInvoiceDocumentServiceImpl.java]: right; ah the perils of cut-n-paste

File: work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java (hunk 1 above)
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/service/impl/ContractsGrantsInvoiceCreateDocumentServiceImpl.java]: yes, and in the other places too, thx

File: work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java (hunk 5 above)
Comment (bhutchinson) [inline work/src/org/kuali/kfs/module/ar/document/ContractsGrantsInvoiceDocument.java]: probably - I can fix that too

**Pull request merged**